website/content/docs/agent/telemetry.mdx

```diff
@@ -269,6 +269,57 @@ resources will still work.
 This metric should be monitored to ensure that the license doesn't expire to prevent degradation of functionality.
 
 
+### Bolt DB Performance
+
+| Metric Name                       | Description                                                      | Unit  | Type  |
+| :-------------------------------- | :--------------------------------------------------------------- | :---- | :---- |
+| `consul.raft.boltdb.freelistBytes`                  | Represents the number of bytes necessary to encode the freelist metadata. When [`raft_boltdb.NoFreelistSync`](/docs/agent/options#NoFreelistSync) is set to `false` these metadata bytes must also be written to disk for each committed log. | bytes | gauge   |
+| `consul.raft.boltdb.logsPerBatch`                   | Measures the number of logs being written per batch to the db. | logs | sample |
+| `consul.raft.boltdb.storeLogs`                      | Measures the amount of time spent writing logs to the db. | ms | timer |
+
+
+**Why they're important:**
+
+The `consul.raft.boltdb.storeLogs` metric is a direct indicator of disk write performance of a Consul server. If there are issues with the disk or
+performance degradations related to Bolt DB, these metrics will show the issue and potentially the cause as well.
+
+**What to look for:**
+
+The primary thing to look for are increases in the `consul.raft.boltdb.storeLogs` times. Its value will directly govern an 
+upper limit to the throughput of write operations within Consul.
+
+In Consul each write operation will turn into a single Raft log to be committed. Raft will process these
+logs and store them within Bolt DB in batches. Each call to store logs within Bolt DB is measured to record how long
+it took as well as how many logs were contained in the batch. Writing logs is this fashion is serialized so that
+a subsequent log storage operation can only be started after the previous one completed. Therefore the maximum number
+of log storage operations that can be performed each second can be calculated with the following equation: 
+`(1000 ms) / (consul.raft.boltdb.storeLogs ms/op)`. From there we can extrapolate the maximum number of Consul writes
```

Comment:
This is totally out of scope for this as *documentation* work, but I admit I read this and wondered why this calculation wasn't surfaced as it's own metric instead of leaving it as an exercise for the user. Something like 'fraction of expected maximum capacity used' or the like?

Reply:
I did manage to craft something that could work for this. I think it ends up giving a rough idea of write capacity and I will be pr'ing to raft-boltdb soon. Once we pull that into Consul then we should probably update these docs further.

The metric isn't exactly 1:1 here as averages get applied at different locations but I think it comes out to the same thing. The metric would be:

```
metrics.AddSample([]string{"raft", "boltdb", "writeCapacity"}, (float32(1000000000) / float32(time.Since(ts).Nanoseconds())) * float32(numLogs))
```

For each `StoreLogs` call to the raft-boltdb storage layer, this will calculate a theoretical write capacity that boltdb could handle based on the number of logs in this batch and how many of these operations could be performed in 1s assuming nothing else slowed it down.

Reply:
I added the aggregate metric to raft-boltdb.